packages/eslint-plugin-office-addins/src/utils/propertiesType.ts

```diff
@@ -0,0 +1,30 @@
+import * as propertiesTypeJson from "./data/propertiesType.json";
+
+export enum PropertyType {
+  navigational,
+  scalar,
+  undefined, // Can be scalar or navigational. Depends of the context.
+  notProperty,
+}
+
+const navigationProperties: Set<string> = new Set<string>(
+  propertiesTypeJson.navigational
+);
+const scalarProperties: Set<string> = new Set<string>(
+  propertiesTypeJson.scalar
+);
+const undefinedProperties: Set<string> = new Set<string>(
```

Comment:
What is an undefined property?

Reply:
Undefined property is when you can't determine if a property is navigational or scalar just by the property name.
For example, the property `format` can be both navigational or scalar, depending on who called it.

I felt like it would add a lot of extra complexity to handle those cases and it would not be worth it, as we only have 9 "undefined" properties, but more than 550 Scalar or navigational.

Reply:
In that case might say "ambiguous" which suggests it's unclear instead of "undefined" which sounds like something that doesn't exist.

Reply:
Makes sense, I do agree it would be a better name (ambiguous), because that's the case